Remove generic list from npm (#504)
kyma-project/console: components/angular/generic-list/README.md

# Generic List Component

This is a private package, not published to npm anymore. We use classical symlinks for usage in the components.

## Development server

Run `npm run serve` for a dev server. Navigate to `http://localhost:4200/`. The app will automatically reload if you change any of the source files.
You will see a sandbox app that uses <y-generic-table> & <y-generic-list> components from the embedded list.module.ts.

## How to use it in a component

> 

Reviewer: Maybe we can add an introductory sentence here, like:
`To use the package in a component, perform the following steps`

and then we may have it as a list of steps. If you don't like the steps, I can rephrase it and we may have a bullet point list. Let me add the steps just to show you the idea.

> Add `--preserve-symlink` option to `ng serve`, `ng test` and `ng build`. 

Author: not in npm cmd anymore, moved to angular.json as config option

Author: done

> Add `"preserveSymlinks": true` to `angular.json` test and build options. Or alternatively add `--preserve-symlink` option to `ng test` and `ng build`. 

Reviewer: ```suggestion
1. Add `"preserveSymlinks": true` to the test and build definitions in the [`angular.json`](https://github.com/kyma-project/console/blob/master/core/angular.json) file. Alternatively,  add `--preserve-symlink` option to `ng test` and `ng build` commands.
```

Reviewer: I added a link to the file - but if you feel it's redundant, we can just reference the file.

Author: Since it is a generic _how to use it_ we should not mix up with a specific component. Although the core/angular.json could be used as reference file. In the end the user needs to edit his very own angular.json from his component.

> Add `"include": [ /components/angular/generic-list/src/app/modules/list/**/*.ts" ]` to your component's `tsconfig.json`

Reviewer: ```suggestion
2. Add `"include": [ /components/angular/generic-list/src/app/modules/list/**/*.ts" ]` to  the `tsconfig.json` file of your component.
```

> Add `"app"` to `tslint.json` if you have the following rules: 

Reviewer: ```suggestion
3. Add `"app"` to  the `tslint.json` file if it includes the following rules: 
```

    "no-submodule-imports": [true, "app"],
    "no-implicit-dependencies": [true, ["app"]],

> Add a symlink to `components/angular/generic-list/src/app/modules/list` inside your application. 

Reviewer: ```suggestion
4. Add a symlink to `components/angular/generic-list/src/app/modules/list` inside your application. 
```

> Example: 

Reviewer: ```suggestion
For example: 
```

> In case of core, we have set the following symlink: 

Reviewer: ```suggestion
Console core has the following symlink set:
```

> `core/src/app/generic-list > components/angular/generic-list/src/app/modules/list` and use it in our Angular application with `import { Filter } from 'app/generic-list'`

Reviewer: I would go with the symlink in a code block and then a sentence. Couldn't do it in a suggestion, too many code blocks :D
````
core/src/app/generic-list > components/angular/generic-list/src/app/modules/list
````
It is then used in the Angular application with `import { Filter } from 'app/generic-list'`.

# Generic List Components

This is a private package, not published to npm anymore. We use symlinks to use it in our Angular microfrontends.

## Development server

Run `npm run serve` for a development server. Go to `http://localhost:4200/`. The application reloads automatically if you change any of the source files.  

> You will see a sandbox app that uses <y-generic-table> & <y-generic-list> components from the embedded list.module.ts.

Reviewer: This one is not a part of the review (explicitly) but maybe we can change it to:
````
As sandbox application appears. It uses <y-generic-table> and <y-generic-list> components from the embedded `list.module.ts` file.

## How to use it in a component

> To use the package in an component, perform the following steps.

Reviewer: ```suggestion
To use the package in a component, perform the following steps:
```

1. Add `"preserveSymlinks": true` to `angular.json` test and build options. Or alternatively add `--preserve-symlink` option to `ng test` and `ng build`. 

2. Add `"include": [ /components/angular/generic-list/src/app/modules/list/**/*.ts" ]` to  the `tsconfig.json` file of your component.

3. Add `"app"` to  the `tslint.json` file if it includes the following rules: 

>     "no-submodule-imports": [true, "app"],

Reviewer: ```suggestion
   * "no-submodule-imports": [true, "app"],
```

>     "no-implicit-dependencies": [true, ["app"]],

Reviewer: ```suggestion
   * "no-implicit-dependencies": [true, ["app"]],
```

4. Add a symlink to `components/angular/generic-list/src/app/modules/list` inside your application.  

> For example:  

Reviewer: ```suggestion
 
```
Combined it with another part.

> Console core has the following symlink set:

Reviewer: ```suggestion
For example, the console core has the following symlink set:
```

>     core/src/app/generic-list > components/angular/generic-list/src/app/modules/list

Reviewer: Let's add it in the codeblock:

````
    core/src/app/generic-list > components/angular/generic-list/src/app/modules/list
````

> 5. Use generic list with `import * as GerericList from 'app/generic-list'`

Reviewer: ```suggestion
5. Use the generic list in `import * as GenericList from 'app/generic-list'` command.
```

Shouldn't it be `GenericList` instead of `GerericList`?
